webpack/assets/javascripts/react_app/components/common/Slot/Slot.js

```diff
@@ -0,0 +1,44 @@
+import { cloneElement, isValidElement } from 'react';
+import PropTypes from 'prop-types';
+import { selectMaxComponent, selectComponentByWeight } from './SlotSelectors';
+
+const Slot = ({ fills, id, multi, children, ...props }) => {
+  const addProps = object => {
+    if (multi && !object.key) {
```

Comment:
Is `key` needed only because `Each child in a list should have a unique "key" prop` or do we use it somewhere directly?

Reply:
I believe the main thing is the `each child` thing, but that is fairly important as without it React is unable to do its mojo and is loosing a lot of performance tweeks. So I would agree there should be an warning about that - the react one doesn't need to kick in as I can develop my plugin in isolation.
Maybe console warning would be sufficient though?

Reply:
Yes, the key is needed due to react architecture, and I agree about console warning in this case :+1:

Reply:
Would it make sense to make `Slot` responsible for creating the keys somehow to make sure they are unique? My thinking is that a developer needs to know about all the places that use a particular slot to make sure the value for a key is really unique, which is really difficult when I consider just the number of 'official' plugins...

Reply:
I think that a component should be responsible for its key (I prefer a domain related key than a random key), if the key isn't unique, then a warning will appear at the console, so a during the developing or reviewing it wouldn't be difficult to verify

Reply:
I guess, I was just wondering if we could make it as easy as possible to use, but maybe adding a note to the docs explaining the key should be unique and there will be warning if it is not will be enough.